base/array.jl

```diff
@@ -1129,6 +1129,21 @@ function pop!(a::Vector)
     return item
 end
 
+function pop!(a::Vector, i::Integer)
+    x = a[i]
+    _deleteat!(a, i, 1);
+    x
```

Comment:
Maybe
```suggestion
    _deleteat!(a, i, 1)
    return x
```
?

Reply:
Ok, I will make a compromise and remove the useless `;` while not adding `return` ;-) 
Adding `return` in long function form is not yet officially recommended and `Base` is full of cases without `return`, so I will contribute to support this lovely style here :)